crates/dojo-core/src/resource_metadata.cairo

```diff
@@ -42,16 +83,29 @@ impl ResourceMetadataModel of dojo::model::Model<ResourceMetadata> {
     }
 
     #[inline(always)]
-    fn layout(self: @ResourceMetadata) -> Span<u8> {
+    fn layout() -> ModelLayout {
         let mut layout = core::array::ArrayTrait::new();
-        dojo::database::introspect::Introspect::<ResourceMetadata>::layout(ref layout);
-        core::array::ArrayTrait::span(@layout)
+        dojo::database::introspect::Introspect::<ResourceMetadata>::layout(ref layout);        
+        ModelLayout::Serial(core::array::ArrayTrait::span(@layout))
```

Comment:
Ideally, we want `ResourceMetadata` to use a `Hash` layout, to unlock the use of `ByteArray`.
Actually.. this `ByteArray` is quite annoying as it's first an array of felts, and then always 2 additional felts. So we may check how this could be handled by the `Hash` layout.

Reply:
Yes, good remark 👍 
I will move `ResourceData` to `hash` layout once it will be fully functional. For now, I keep it as `serial` to have tests green.